### DISCUSSION TITLE
State of Flash Encryption and Secure Boot for ESP32

### DISCUSSION THREAD

**Discussion by git-n-pissed** [micropython/micropython#10530] [ESP32]
I'm trying to figure out the state of [Flash Encryption](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/security/flash-encryption.html) and [Secure Boot V2](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/security/secure-boot-v2.html) for the ESP32.  I would do some experimenting, but since such experimentation would require burning efuses, I wanted to see if anyone knew where things currently stand before I start making bricks.

[Here](https://github.com/micropython/micropython/issues/8706) is an issue related to Flash Encryption, but the OP's original question has not been answered.

[Here](https://github.com/micropython/micropython/issues/4854) is an issue to Secure Boot, and @dpgeorge did respond, but the last post was over 3 years ago :(

**Comment by stephanelsmith** [comment]
This is coming to head for me as well, as there are some items in my firmware needing protection.

In addition to flash encryption and secure boot, the other item are the [UART ROM Download modes](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/kconfig.html#config-secure-uart-rom-dl-mode).

On the micropython side, there's "Disabling CTRL-C / REPL" consideration in your last link from long ago.
